### PR TITLE
Build - clean warnings and errors

### PR DESCRIPTION
**Pull request by rdlopes**
A series of refactorings and fixes to remove warnings and errors caused by build and runtime setup.

### CODE REVIEW

**Comment by rdlopes** [comment]
This one is ready for a bump to Java 11 and a clean!

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/api/DocumentAttributes.java [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
 import java.util.Map;
-import java.util.logging.Level;
-import java.util.logging.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Created by pestano on 04/06/15. Document overall configuration */
 public class DocumentAttributes {
 
+  public static final Logger log = LoggerFactory.getLogger(DocumentAttributes.class);
```
Comment (rmpestano) [inline cukedoctor-converter/src/main/java/com/github/cukedoctor/api/DocumentAttributes.java:17]: Can we make it private?

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/api/model/Step.java [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
 import com.fasterxml.jackson.annotation.JsonRawValue;
 import java.util.List;
-import java.util.logging.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Step {
 
+  public static final Logger log = LoggerFactory.getLogger(Step.class);
```
Comment (rmpestano) [inline cukedoctor-converter/src/main/java/com/github/cukedoctor/api/model/Step.java:15]: Same here, can we make it private?

File: cukedoctor-extension/src/main/java/com/github/cukedoctor/extension/CukedoctorFooterExtension.java [hunk 3]
```diff
@@ -2,15 +2,18 @@
...
 import org.asciidoctor.extension.Postprocessor;
 import org.jsoup.Jsoup;
 import org.jsoup.nodes.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Created by pestano on 20/07/15. adds search box to rendered html documentation */
 public class CukedoctorFooterExtension extends Postprocessor {
 
+  public static final Logger log = LoggerFactory.getLogger(CukedoctorFooterExtension.class);
```
Comment (rmpestano) [inline cukedoctor-extension/src/main/java/com/github/cukedoctor/extension/CukedoctorFooterExtension.java:15]: Let's make it private

**Review by rmpestano — CHANGES_REQUESTED** [review]

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/api/DocumentAttributes.java (hunk 1 above)
Comment (rdlopes) [reply cukedoctor-converter/src/main/java/com/github/cukedoctor/api/DocumentAttributes.java]: Should have been. Done.

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/api/model/Step.java (hunk 2 above)
Comment (rdlopes) [reply cukedoctor-converter/src/main/java/com/github/cukedoctor/api/model/Step.java]: Should have been. Done.

File: cukedoctor-extension/src/main/java/com/github/cukedoctor/extension/CukedoctorFooterExtension.java (hunk 3 above)
Comment (rdlopes) [reply cukedoctor-extension/src/main/java/com/github/cukedoctor/extension/CukedoctorFooterExtension.java]: Should have been. Done.

**Comment by rmpestano** [comment]
> This one is ready for a bump to Java 11 and a clean!

Thank you so much for this @rdlopes! After merging this I think we can release 3.9.0 and then upgrade to use Java 11 as base and start working on version 4.x. 

One thing that we were discussing in the past was the Antora integration (#178), I think Antora could replace the discontinued Jenkins plugin (#157) as a standard/recommended way for publishing the living docs.

**Review by rmpestano — APPROVED** [review]

**Pull request merged**